packages/snjs/lib/services/api/mfa_service.ts

```diff
@@ -0,0 +1,169 @@
+import { Challenge, ChallengeKeyboardType, ChallengePrompt, ChallengeReason, ChallengeValidation } from '@Lib/challenges';
+import { ChallengeService, SNAlertService } from '..';
+import { PureService } from '../pure_service';
+import { SNApiService } from './api_service';
+import { PromptTitles, SessionStrings, SignInStrings } from './messages';
+import { HttpResponse, StatusCode } from './responses';
+
+export enum MfaStatus {
+  EnteredCode = 'mfa-entered-code',
+  NotRequired = 'mfa-not-required',
+  Ignored = 'mfa-ignored',
+
+  ErrorMissingCode = 'mfa-error-missing-code',
+  ErrorResponse = 'mfa-error-response',
+}
+
+export type MfaResultOld = {
+  status: MfaStatus.EnteredCode,
+  mfaKey: string,
+  enteredMfaCode: string,
+} | {
+  status: Exclude<MfaStatus, MfaStatus.EnteredCode>,
+  response: HttpResponse<unknown>,
+}
+
+export type MfaResult = {
+  status: MfaStatus.EnteredCode,
+  totpUuid: string,
+  enteredTotp: string,
+} | {
+  status: MfaStatus.ErrorResponse,
+  response: HttpResponse,
+} | {
+  status: Exclude<MfaStatus, 
+    MfaStatus.EnteredCode | 
+    MfaStatus.ErrorResponse
+  >,
+}
+
+
+// todo: figure out if and where this belongs
+// possibly this belongs in the Auth SDK and it should be shared with Auth
+type GetAuthMethodsResponse = {
```

Comment:
This is the type of the response from the new /auth/methods endpoint copied from Auth. It's a part of a common interface which seems to live in the auth package (Auth SDK) in SNJS -- so it most likely belongs there.